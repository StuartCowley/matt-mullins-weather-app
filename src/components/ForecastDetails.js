import React from "react";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <h3 className="forecast-details__date">{formattedDate}</h3>
      <div className="forecast-details__max-temp">
        Maximum temperature: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__min-temp">
        Minimum temperature: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__wind-speed">
        Wind speed: {wind.speed}mph
      </div>
      <div className="forecast-details__wind-direction">
        Wind direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;