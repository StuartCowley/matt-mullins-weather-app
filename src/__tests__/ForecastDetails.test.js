import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
    const validProps ={
        date: 1525046400000,
        temperature: {
            min: 12,
            max: 22,
        },
        wind: {
            speed: 13,
            direction: "5",
        },
        humidity: 30,
    };

    it("renders correctly", () => {
        const { asFragment } = render(<ForecastDetails forecast={validProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
});