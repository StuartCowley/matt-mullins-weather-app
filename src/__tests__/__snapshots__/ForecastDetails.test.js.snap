// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <h3
      class="forecast-details__date"
    >
      Mon Apr 30 2018
    </h3>
    <div
      class="forecast-details__max-temp"
    >
      Maximum temperature: 22°C
    </div>
    <div
      class="forecast-details__min-temp"
    >
      Minimum temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind speed: 13mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind direction: 5
    </div>
  </div>
</DocumentFragment>
`;
